src/main/java/io/github/moulberry/notenoughupdates/util/ApiUtil.java

```diff
@@ -139,23 +140,24 @@ public Request postData(String contentType, String data) {
 			return this;
 		}
 
-		private CompletableFuture<URL> buildUrl() {
-			CompletableFuture<URL> fut = new CompletableFuture<>();
+		public CompletableFuture<String> requestString() {
+			CompletableFuture<String> result = new CompletableFuture<>();
+			URL url;
 			try {
-				fut.complete(new URIBuilder(baseUrl)
-					.addParameters(queryArguments)
-					.build()
-					.toURL());
-			} catch (URISyntaxException |
-							 MalformedURLException |
-							 NullPointerException e) { // Using CompletableFuture as an exception monad, isn't that exiting?
-				fut.completeExceptionally(e);
+				url = new URIBuilder(baseUrl).addParameters(queryArguments).build().toURL();
+			} catch (URISyntaxException | MalformedURLException | NullPointerException e) {
+				String message = "Error while building url! baseUrl: '" + this.baseUrl + "',  queryArguments: " + queryArguments;
+				result.completeExceptionally(new RuntimeException(message, e));
+				return result;
 			}
-			return fut;
-		}
+			String link = url.toString();
 
-		public CompletableFuture<String> requestString() {
-			return buildUrl().thenApplyAsync(url -> {
+			CompletableFuture<String> cachedResult = HypixelAPICache.INSTANCE.getCacheFor(link);
+			if (cachedResult != null) return cachedResult;
+
+			HypixelAPICache.INSTANCE.addToCache(link, result);
```

Comment:
You should not cache POST request (as per [RFC2616](https://www.rfc-editor.org/rfc/rfc2616.html#section-9.5))

Reply:
afaik all requesrs are post. so no caching at all? idk how the rfc text is relevant to caching in general. your pr does cache everything too sooo?

Reply:
most requests are GET. POST requests are for changing data and therefore should not be cached. my pr does not cache POST requests (if ("GET".equals(method)) cacheKey = null). the rfc is just saying that every http implementation should not cache post requests unless there are headers specified that permit caching (we do not handle cache control headers at all, so that doesnt really matter)

Reply:
setting cape